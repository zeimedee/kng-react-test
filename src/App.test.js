
import {render, unmountComponentAtNode} from "react-dom";
import {act} from "react-dom/test-utils"
import Radio from './radio'

let container = null;

beforeEach(()=>{
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(()=>{
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("changes color when clicked", ()=>{
    act(()=>{
      render(<Radio />, container)
    });
    expect(container.textContent).toBe("text");
    const radio1 = document.querySelector("#rad1");
    const radio2 = document.querySelector("#rad2");

    act(()=>{
      radio1.dispatchEvent(new MouseEvent("click", { bubbles: true}));
    });
    expect(container).toMatchSnapshot();
    act(()=>{
      radio2.dispatchEvent(new MouseEvent("Click", {bubbles: true}));
    })
    expect(container).toMatchSnapshot();


    
})
