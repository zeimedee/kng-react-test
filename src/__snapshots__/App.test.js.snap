// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes color when clicked 1`] = `
<div>
  <div>
    <h1
      id="text"
      style="color: red;"
    >
      text
    </h1>
    <input
      checked=""
      id="rad1"
      name="b1"
      type="radio"
    />
    <input
      id="rad2"
      name="b2"
      type="radio"
    />
  </div>
</div>
`;

exports[`changes color when clicked 2`] = `
<div>
  <div>
    <h1
      id="text"
      style="color: red;"
    >
      text
    </h1>
    <input
      checked=""
      id="rad1"
      name="b1"
      type="radio"
    />
    <input
      id="rad2"
      name="b2"
      type="radio"
    />
  </div>
</div>
`;
